homeassistant/components/bluesound/media_player.py

```diff
@@ -619,22 +613,36 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
 
     def rebuild_bluesound_group(self) -> list[str]:
         """Rebuild the list of entities in speaker group."""
-        if self._group_name is None:
+        if self.sync_status.master is None and self.sync_status.slaves is None:
             return []
 
-        device_group = self._group_name.split("+")
+        player_entities: list[BluesoundPlayer] = self.hass.data[DATA_BLUESOUND]
+
+        leader_sync_status: SyncStatus | None = None
+        match self.sync_status.master:
+            case None:
+                leader_sync_status = self.sync_status
+            case paired_player:
+                leader_sync_status_list = [
+                    x.sync_status
+                    for x in player_entities
+                    if x.sync_status.id == f"{paired_player.ip}:{paired_player.port}"
+                ]
+                if leader_sync_status_list:
+                    leader_sync_status = leader_sync_status_list[0]
+
+        if leader_sync_status is None or leader_sync_status.slaves is None:
+            return []
 
-        sorted_entities: list[BluesoundPlayer] = sorted(
-            self.hass.data[DATA_BLUESOUND],
-            key=lambda entity: entity.is_master,
-            reverse=True,
-        )
-        return [
-            entity.sync_status.name
-            for entity in sorted_entities
-            if entity.bluesound_device_name in device_group
+        follower_ids = [f"{x.ip}:{x.port}" for x in leader_sync_status.slaves]
+        follower_names = [
+            x.sync_status.name
+            for x in player_entities
+            if x.sync_status.id in follower_ids
         ]
 
+        return [leader_sync_status.name, *follower_names]
```

Comment:
```suggestion
        names = [leader_sync_status.name]
        for x in player_entities:
            if x.sync_status.id in follower_ids:
                names.append(x.sync_status.name)

        return names
```

Reply:
Ruff doesn't like the `names.append(...)` and complains with [PERF401](https://docs.astral.sh/ruff/rules/manual-list-comprehension/). It boils down to: "Use a list comprehension instead`.